kombu/transport/SQS.py

```diff
@@ -154,10 +154,10 @@ def entity_name(self, name, table=CHARS_REPLACE_TABLE):
         """Format AMQP queue name into a legal SQS queue name."""
         if name.endswith('.fifo'):
             partial = name.rstrip('.fifo')
-            partial = text_t(safe_str(partial)).translate(table)
+            partial = text_t(safe_str(partial.translate(table)))
```

Comment:
Could you please a very simple test case to verify this change? You already have an example in https://github.com/celery/kombu/issues/791 , so you could use that.